holidays/countries/niger.py

```diff
@@ -0,0 +1,314 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import (
+    JAN,
+    FEB,
+    MAR,
+    APR,
+    MAY,
+    JUN,
+    JUL,
+    AUG,
+    SEP,
+    OCT,
+    NOV,
+    DEC,
+    SUN,
+)
+from holidays.groups import ChristianHolidays, InternationalHolidays, IslamicHolidays
+from holidays.observed_holiday_base import ObservedHolidayBase, SUN_TO_NEXT_MON
+
+
+class Niger(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays):
+    """Niger holidays.
+
+    References:
+        * [Law No. 59-22 of December 24, 1959](https://web.archive.org/web/20241106023958/https://www.impots.gouv.ne/media/loi/1960.pdf)
+        * <https://web.archive.org/web/20110721063839/http://www.ais-asecna.org/pdf/gen/gen-2-1/12gen2-1-01.pdf>
+        * <https://wageindicator.org/documents/decentworkcheck/africa/niger-french-2021.pdf>
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Niger>
+        * <https://wageindicator.org/fr-ne/droit-du-travail/les-conges-et-les-jours-de-repos>
+        * <https://web.archive.org/web/20250206194155/https://www.rivermate.com/guides/niger/leave>
+        * [Eid al-Adha](https://web.archive.org/web/20250117013558/https://www.timeanddate.com/holidays/niger/eid-al-adha)
+        * [Eid al-Fitr](https://web.archive.org/web/20250114130118/https://www.timeanddate.com/holidays/niger/eid-al-fitr)
+        * [Laylat al-Qadr](https://www.timeanddate.com/holidays/niger/lailat-al-qadr)
```

Comment:
Minor link typo here
```suggestion
        * [Laylat al-Qadr](https://web.archive.org/web/20250531033137/https://www.timeanddate.com/holidays/niger/laylat-al-qadr)
```